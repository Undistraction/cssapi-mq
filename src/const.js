// -----------------------------------------------------------------------------
// Exports
// -----------------------------------------------------------------------------

export const MEDIA_TYPES = Object.freeze({
  ALL: 'all',
  PRINT: 'print',
  SCREEN: 'screen',
  SPEECH: 'speech',
  NONE: '',
});

export const DIMENSIONS_UNITS = Object.freeze({
  EM: 'em',
  REM: 'rem',
  PX: 'px',
});

export const UNITS = Object.freeze({
  DIMENSIONS: DIMENSIONS_UNITS,
  RESOLUTION: Object.freeze({ DPI: 'dpi' }),
});

export const BREAKPOINT_MAP_NAMES = Object.freeze([
  'width',
  'height',
  'resolution',
]);

export const ORIENTATIONS = Object.freeze(['portrait', 'landscape']);

export const SEPARATOR_VALUES = Object.freeze({
  rem: 0.01,
  em: 0.01,
  px: 1,
  dpi: 1,
});

export const MEDIA_PREFIX = '@media';
