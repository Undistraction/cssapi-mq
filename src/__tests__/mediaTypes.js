import { values, drop } from 'ramda';
import { MEDIA_TYPES } from '../const';
import { mqWithValidBreakpointsForRange } from './data';
import cssSerialiser from './helpers/cssSerialiser';

expect.addSnapshotSerializer(cssSerialiser);
const validValues = values(MEDIA_TYPES);

describe('mediaTypes', () => {
  it('returns the correct default media type if called with no arguments', () => {
    expect(
      mqWithValidBreakpointsForRange('width').mediaType()
    ).toMatchSnapshot();
  });

  for (const value of validValues) {
    it(`returns the supplied mediaTypes for '${value}'`, () => {
      expect(
        mqWithValidBreakpointsForRange('width').mediaType(value)
      ).toMatchSnapshot();
    });
  }

  it('supports multiple values', () => {
    expect(
      mqWithValidBreakpointsForRange('width').mediaType(drop(2, validValues))
    ).toMatchSnapshot();
  });

  it('throws if argument is not valid media type', () => {
    expect(() =>
      mqWithValidBreakpointsForRange('width').mediaType('xxxx')
    ).toThrowErrorMatchingSnapshot();
  });
});
