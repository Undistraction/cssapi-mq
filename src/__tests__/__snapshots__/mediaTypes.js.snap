// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mediaTypes returns default media type if called with an empty array 1`] = `screen`;

exports[`mediaTypes returns default media type if called with no arguments 1`] = `screen`;

exports[`mediaTypes returns the correct default media type if called with no arguments 1`] = `screen`;

exports[`mediaTypes returns the supplied mediaTypes for 'all' 1`] = `all`;

exports[`mediaTypes returns the supplied mediaTypes for 'print' 1`] = `print`;

exports[`mediaTypes returns the supplied mediaTypes for 'screen' 1`] = `screen`;

exports[`mediaTypes returns the supplied mediaTypes for 'speech' 1`] = `speech`;

exports[`mediaTypes supports multiple values 1`] = `screen, speech`;

exports[`mediaTypes throws if argument is '' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: ''.`;

exports[`mediaTypes throws if argument is '' 2`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: ''.`;

exports[`mediaTypes throws if argument is '[object Object]' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: '[object Object]'.`;

exports[`mediaTypes throws if argument is '444' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: '444'.`;

exports[`mediaTypes throws if argument is 'false' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: 'false'.`;

exports[`mediaTypes throws if argument is 'true' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: 'true'.`;

exports[`mediaTypes throws if argument is 'xxxx' 1`] = `'mediaType'must be one of 'all, print, screen, speech' but you supplied: 'xxxx'.`;
